File: clients/Java-GraalVM-native/src/main/java/model/DebugCommand.java
package model;

import util.StreamUtil;

public abstract class DebugCommand {
    public abstract void writeTo(java.io.OutputStream stream) throws java.io.IOException;
    public static DebugCommand readFrom(java.io.InputStream stream) throws java.io.IOException {
        switch (StreamUtil.readInt(stream)) {
            case Add.TAG:
                return Add.readFrom(stream);
            case Clear.TAG:
                return Clear.readFrom(stream);
            case SetAutoFlush.TAG:
                return SetAutoFlush.readFrom(stream);
            case Flush.TAG:
                return Flush.readFrom(stream);
            default:
                throw new java.io.IOException("Unexpected tag value");
        }
    }

    public static class Add extends DebugCommand {
        public static final int TAG = 0;
        private model.DebugData data;
        public model.DebugData getData() { return data; }
        public void setData(model.DebugData data) { this.data = data; }
        public Add() {}
        public Add(model.DebugData data) {
            this.data = data;
        }
        public static Add readFrom(java.io.InputStream stream) throws java.io.IOException {
            Add result = new Add();
            result.data = model.DebugData.readFrom(stream);
            return result;
        }
        @Override
        public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
            StreamUtil.writeInt(stream, TAG);
            data.writeTo(stream);
        }
    }

    public static class Clear extends DebugCommand {
        public static final int TAG = 1;
        public Clear() {}
        public static Clear readFrom(java.io.InputStream stream) throws java.io.IOException {
            Clear result = new Clear();
            return result;
        }
        @Override
        public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
            StreamUtil.writeInt(stream, TAG);
        }
    }

    public static class SetAutoFlush extends DebugCommand {
        public static final int TAG = 2;
        private boolean enable;
        public boolean isEnable() { return enable; }
        public void setEnable(boolean enable) { this.enable = enable; }
        public SetAutoFlush() {}
        public SetAutoFlush(boolean enable) {
            this.enable = enable;
        }
        public static SetAutoFlush readFrom(java.io.InputStream stream) throws java.io.IOException {
            SetAutoFlush result = new SetAutoFlush();
            result.enable = StreamUtil.readBoolean(stream);
            return result;
        }
        @Override
        public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
            StreamUtil.writeInt(stream, TAG);
            StreamUtil.writeBoolean(stream, enable);
        }
    }

    public static class Flush extends DebugCommand {
        public static final int TAG = 3;
        public Flush() {}
        public static Flush readFrom(java.io.InputStream stream) throws java.io.IOException {
            Flush result = new Flush();
            return result;
        }
        @Override
        public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
            StreamUtil.writeInt(stream, TAG);
        }
    }
}

File: clients/Kotlin-GraalVM-native/src/main/kotlin/model/DebugCommand.kt
package model

import util.StreamUtil

abstract class DebugCommand {
    @Throws(java.io.IOException::class)
    abstract fun writeTo(stream: java.io.OutputStream)
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): DebugCommand {
            when (StreamUtil.readInt(stream)) {
                Add.TAG -> return Add.readFrom(stream)
                Clear.TAG -> return Clear.readFrom(stream)
                SetAutoFlush.TAG -> return SetAutoFlush.readFrom(stream)
                Flush.TAG -> return Flush.readFrom(stream)
                else -> throw java.io.IOException("Unexpected tag value")
            }
        }
    }

    class Add : DebugCommand {
        lateinit var data: model.DebugData
        constructor() {}
        constructor(data: model.DebugData) {
            this.data = data
        }
        companion object {
            val TAG = 0
            @Throws(java.io.IOException::class)
            fun readFrom(stream: java.io.InputStream): Add {
                val result = Add()
                result.data = model.DebugData.readFrom(stream)
                return result
            }
        }
        @Throws(java.io.IOException::class)
        override fun writeTo(stream: java.io.OutputStream) {
            StreamUtil.writeInt(stream, TAG)
            data.writeTo(stream)
        }
    }

    class Clear : DebugCommand {
        constructor() {}
        companion object {
            val TAG = 1
            @Throws(java.io.IOException::class)
            fun readFrom(stream: java.io.InputStream): Clear {
                val result = Clear()
                return result
            }
        }
        @Throws(java.io.IOException::class)
        override fun writeTo(stream: java.io.OutputStream) {
            StreamUtil.writeInt(stream, TAG)
        }
    }

    class SetAutoFlush : DebugCommand {
        var enable: Boolean = false
        constructor() {}
        constructor(enable: Boolean) {
            this.enable = enable
        }
        companion object {
            val TAG = 2
            @Throws(java.io.IOException::class)
            fun readFrom(stream: java.io.InputStream): SetAutoFlush {
                val result = SetAutoFlush()
                result.enable = StreamUtil.readBoolean(stream)
                return result
            }
        }
        @Throws(java.io.IOException::class)
        override fun writeTo(stream: java.io.OutputStream) {
            StreamUtil.writeInt(stream, TAG)
            StreamUtil.writeBoolean(stream, enable)
        }
    }

    class Flush : DebugCommand {
        constructor() {}
        companion object {
            val TAG = 3
            @Throws(java.io.IOException::class)
            fun readFrom(stream: java.io.InputStream): Flush {
                val result = Flush()
                return result
            }
        }
        @Throws(java.io.IOException::class)
        override fun writeTo(stream: java.io.OutputStream) {
            StreamUtil.writeInt(stream, TAG)
        }
    }
}
